packages/across-monitor/src/AcrossMonitor.ts

```diff
@@ -0,0 +1,151 @@
+    } else {
+      this.startingBlock = this.endingBlock ? this.endingBlock + 1 : latestL1BlockNumber;
+      this.endingBlock = latestL1BlockNumber;
+    }
+    // Starting block should not be after the ending block (this could happen on short polling period or misconfiguration).
+    this.startingBlock = Math.min(this.startingBlock, this.endingBlock);
+
+    await this.relayEventProcessor.update(this.endingBlock);
+  }
+
+  async checkUtilization(): Promise<void> {
+    this.logger.debug({ at: "AcrossMonitor#Utilization", message: "Checking for pool utilization ratio" });
+
+    // Collect utilization and other properties for all bridge pools.
+    const bridgePools = await Promise.all(
+      Object.keys(this.bridgePools).map(async (l1TokenAddress) => {
+        const utilization = await this.bridgePools[l1TokenAddress].contract.methods
+          .liquidityUtilizationCurrent()
+          .call();
+        return {
+          address: this.bridgePools[l1TokenAddress].contract.options.address,
+          chainId: this.monitorConfig.bridgeAdminChainId,
+          poolCollateralSymbol: this.bridgePools[l1TokenAddress].poolCollateralSymbol,
+          utilization: utilization,
+        };
+      })
+    );
+
+    // Send notification if pool utilization is above configured threshold.
+    for (const bridgePool of bridgePools) {
+      if (
+        toBN(bridgePool.utilization).gt(
+          toBN(this.monitorConfig.utilizationThreshold)
+            .mul(toBN(toWei("1")))
+            .div(toBN(100))
+        )
+      ) {
+        this.logger.warn({
+          at: "UtilizationMonitor",
+          message: "Across bridge pool utilization warning⚠",
+          mrkdwn:
+            bridgePool.poolCollateralSymbol +
+            " bridge pool at " +
+            createEtherscanLinkMarkdown(bridgePool.address, bridgePool.chainId) +
+            " on " +
+            PublicNetworks[bridgePool.chainId]?.name +
+            " is at " +
+            createFormatFunction(0, 2)(toBN(bridgePool.utilization).mul(toBN(100))) +
+            "% utilization!",
+        });
+      }
+    }
+
+    return;
+  }
+
+  async checkUnknownRelays(): Promise<void> {
+    this.logger.debug({ at: "AcrossMonitor#UnknownRelays", message: "Checking for unknown relays" });
+
+    const relayEvents: EventInfo[] = await this.relayEventProcessor.getRelayEventInfo(
+      this.startingBlock,
+      this.endingBlock
+    );
+    for (const event of relayEvents) {
+      // Skip notifications for known relay caller addresses.
+      if (this.monitorConfig.whitelistedAddresses.includes(event.caller.toLowerCase())) {
+        continue;
+      }
+
+      this.logger.warn({
+        at: "UnknownRelays",
+        message: "Across bridge pool unknown relay event warning⚠",
+        mrkdwn:
+          createEtherscanLinkMarkdown(event.caller, this.monitorConfig.bridgeAdminChainId) +
+          " " +
+          event.action +
```

Comment:
this is awesome. I was wondering how you'd use the `action` prop. this makes sense!